lib/src/main/java/com/github/okdroid/checkablechipview/CheckableChipView.kt

```diff
@@ -366,6 +368,51 @@ class CheckableChipView @JvmOverloads constructor(
         touchFeedbackDrawable.setHotspot(x, y)
     }
 
+    override fun onSaveInstanceState(): Parcelable {
+        return SavedState(super.onSaveInstanceState()).apply {
+            checked = isChecked
+        }
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        val savedState = state as SavedState
+        super.onRestoreInstanceState(savedState.superState)
+
+        val listener = onCheckedChangeListener
+        onCheckedChangeListener = null
+        isChecked = savedState.checked
+        onCheckedChangeListener = listener
+    }
+
+    class SavedState : BaseSavedState {
+
+        var checked: Boolean = false
+
+        constructor(state: Parcelable) : super(state)
+
+        constructor(parcel: Parcel) : super(parcel) {
+            checked = parcel.readInt() == 1
+        }
+
+        override fun describeContents() = 0
+
+        override fun writeToParcel(dest: Parcel, flags: Int) {
+            dest.writeInt(if (checked) 1 else 0)
+        }
+
+        @Suppress("unused")
+        companion object {
+            @JvmField
+            val CREATOR = object : Parcelable.Creator<SavedState> {
+                override fun createFromParcel(source: Parcel): SavedState {
```

Comment:
I'd make it a one-liner as well `override fun createFromParcel(source: Parcel) = SavedState(source)`